From b85ab361b9be1d6a6027dfe35153f35c75d4004a Mon Sep 17 00:00:00 2001
From: Tim Snyder <snyder.tim@gmail.com>
Date: Sat, 9 Apr 2022 17:17:08 +0000
Subject: [PATCH] use PATH resolution when testing dwarfdump

perhaps we haven't built it but we want to test the installed binary
---
 test/dwarfdumpsetup.sh | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/test/dwarfdumpsetup.sh b/test/dwarfdumpsetup.sh
index fcbf8ddbd..2a0fb1cca 100755
--- a/test/dwarfdumpsetup.sh
+++ b/test/dwarfdumpsetup.sh
@@ -45,7 +45,13 @@ else
   textlim=700
 fi
 cp "$top_srcdir/src/bin/dwarfdump/dwarfdump.conf" .
-dd=$top_blddir/src/bin/dwarfdump/dwarfdump
+# prefix PATH with build location of dwarfdump
+# if configure --disable-dwarfdump this will only work
+# tests will only work if dwarfdump is installed somewhere
+# else in the path already and working
+export PATH=$top_blddir/src/bin/dwarfdump:$PATH
+dd=dwarfdump
+echo -n "Testing dwarfdump @ "; which dwarfdump
 # Remove the leading two lines for windows
 # as windows dwarfdump emits two leading lines
 # as compared to non-windows dwarfdump
-- 
2.35.1

